### DISCUSSION TITLE
What to do when your train is too heavy to go up an incline

### DISCUSSION THREAD

**Discussion by TekyTeky** [Novakasa/brickrail#154] [General]
I am having an issue, that I think you said you also have with one of your heavy trains, that is it won’t climb up the incline. I decided I wanted a steam train with old passenger wagons to go to the new station at the top of the incline. But after remodelling the tender, which houses the hub and sensor, the Tain has the motor, it is now too heavy. It will go by itself, just. Is there any provision to adjust the torque somehow? Would changing the acceleration help?
All else is going well. I’ll update to the new master for the keep alive type issue.

This is the incline, then the station. As you can see I have worked a lot on the station and train track. So would like to have it work.

![IMG_2555](https://github.com/Novakasa/brickrail/assets/87248517/d4c51966-96f6-4699-bd4f-bf3f4de51c8f)
![IMG_2557](https://github.com/Novakasa/brickrail/assets/87248517/7a31680b-512d-430d-bd44-deb953f3d8aa)
![IMG_2556](https://github.com/Novakasa/brickrail/assets/87248517/76b9415c-a1b4-4250-951e-313c55709a9f)

**Comment by Novakasa** [comment]
I'm guessing you already use the rubber rings from trixbrix? Are the wheels still slipping or is it just a torque issue?
I do think higher acceleration might help, also putting the first marker as early before the incline as possible might make the train gain more momentum earlier. It probably also works best with fully charged batteries (non-rechargeable 1.5V batteries will probably be better as well)

I'd also experiment with manually controlling the train to see if it is theoretically possible and observe what control pattern is necessary to make the incline. If the train can't make it even manually, then you probably have to modify the gradient or the weight of the train...

For me, I did make the incline as long as possible and made sure that the gradient has smooth vertical bends, so no sharp kinks between the tracks. A straight incline is easier torque-wise as well.

**Reply by Novakasa** [reply]
Also, I'm guessing you have left the "fast speed" at 100 for the train? That will produce the most torque.
However if the wheels are slipping you need to balance acceleration vs. momentum. You want fast enough acceleration so the train has the highes speed at the beginning of the incline, but if you accelerate too fast the wheels might start slipping.

**Reply by TekyTeky** [reply]
OK, I tried all the above but still didn't make any difference... even moving it back to the default Lego FW, and using the remote.
But, I have made some major changes to the track and removed most of the curves.
Also used new 1.5v AAA batteries with Brickrail... but ther are some issues.

Can you explain, please, how the acceleation and deacceleration numbers work?
Coming down the ramp backwards is very fast, even with slow at 35.
Then at the top of the ramp I have a slow sensor but it has so much momentum that it over flies the end of the tyack.. not ideal.

I really want to make this work but it is proving to be difficult.

**Reply by TekyTeky** [reply]
I think i have tricked it, I have the same marker , the first red sensor on 2 blocks. It is on the flat before the incline and is set to fast on the fwd setting... thus it gets momentum. When it is used for the other block, not going up the incline, it is in cruise mode and seems to use that when it is selected to go straight. I also changed the block near the end station and now it goes fast but stops with plenty of room and not go over the end.

**Reply by TekyTeky** [reply]
Also on the battery side, I have read some people use the recghargable Li-Ni 9volt batteries instead of teh 6 AAA.. so I might try that as a better option as I have used up 6 AAA normal batteries just today while testing. It means solcering a 9v lead to the hub.. but I am very good at delecate soldering.. so shouldn't be an issue.

**Reply by TekyTeky** [reply]
So I would still like to know how the Acceleration and Deceleration works... eg higher numbers vs lower numbers.
Because if I set a stop location (Blue sensor in the block) for the steam train, beause of the momentum it stops just right, but the Green train has less momentum and stops directly after the Blue sensor marker.

**Reply by Novakasa** [reply]
so speeds and acceleration only ever control the DC input to the motor. It's probably closer to "torque" than "speed", since the actual physical speed of the train will depend on how heavy the train is and how much friction it has, but it should be proportional to the DC value.

The acceleration/deceleration values control how the DC changes after the train switches states between "stop", "slow", "cruise" and "fast". Each state is associated with a DC value (stop is 0, the rest is what you configure in the train inspector). For example, if you have a "slow" speed of 20, and a "cruise" speed of 50, and the train switches from "slow" to "cruise", the DC it actually puts on the motor changes gradually. If the acceleration is set to 30, then it takes exactly one second for the DC to actually be set to 50. If the acceleration is set to 60, the DC value will reach 50 after already 0.5 seconds.

Same for deceleration, if the train goes from "cruise" to "slow" it will take 1 second to reach the "slow" DC value when the acceleration is set to 30.

This of course does not mean that the train actually reaches the slow speed after 1 second, if the train is very heavy it has physical inertia so the DC of 30 might take a while for the train to actually slow down to a constant speed.

Also, there is the exception that if the train is set to "stop", the DC will always be set to 0 _immediately_. So the deceleration only applies when braking to slow or cruise

**Reply by Novakasa** [reply]
I am not 100% sure I follow how you set up the sensor speed in the end, but if it works that's great! Do you have a suspicion that maybe there's a bug where sometimes the train does not actually change the speed after it encounters a marker?

You also have to keep in mind that the train will "see" the marker much later when it goes backwards, since it only has the sensor in the front.

You can also just add regular markers (does _not_ need to be a prior marker). 
So you can just add a marker to a random track element and assign a speed to it in one direction.

BTW, I've now renamed prior sensors to "backwards entry markers".

**Reply by TekyTeky** [reply]
Ok, got all that..thanks.. using voltage makes more sense to me, and now I know that 30 equals 1 second. 

I had never thought of adding a single marker anywhere to change speed. I guess the distance between blocks on my layout is small enough to use the block markers.
The new steam train has the sensor in the tender so it sees markers differently to the green train. 

I think this isn’t possible but you may have an idea….. Because the steam train has the sensor in the tender I have placed the blue marker so the carriage, being pulled, stops at the platform. But the green train stops too early as the sensor is at the front. As there isn’t enough room for another block,  how can I get it to stop further along the platform?

Yes,  I had figured you had changed the name of the prior sensors 😎

**Reply by Novakasa** [reply]
30 only means 1 second when going from 20 to 50 (50 - 20 = 30)
For example, when going from 20 to 80, it takes 2 seconds (because the speed changes by 30/s, thus it takes 2 seconds to increase by 60)

**Reply by TekyTeky** [reply]
My other question may not have made sense.

Can you have 2 stop positions, one for each train in the same block?

The steam train has the sensor in the tender, the green train is in front.
So I need them to stop at different places on the one block.

Could I have a "mini" block?
As it is at the end, if I have a small block, maybe 2 tracks long past the one the steam train stops at.. it maybe ok, as it won't be used for Random mode yet while I get it all right.

Ok, I have done something great (I think so anyway).. :)

I created 2 blocks next to each other, the main block and a small one towards the end of the track (for the greeen train to stop)
It all seems to work.. the stream train stops at the first blue marker if I select that, the green train stops at the second blue marker if I select that.
Both reverse out ok too.
![82205B4D-DB1C-4AF4-AE81-59804932A5CB_1_201_a](https://github.com/Novakasa/brickrail/assets/87248517/7e5b1e31-09bb-4f9e-b665-990d5a191453)

![2EAB943A-3B77-4B55-BC9E-3289409F9C93_1_201_a](https://github.com/Novakasa/brickrail/assets/87248517/b105e441-acd2-485d-a421-610644306950)

**Reply by Novakasa** [reply]
Interesting solution!

Though this probably has a few weaknesses:
-  in random mode, each train will chose either one of the blocks to stop at. This might be less of an issue once we have the "train schedule" feature though. We could also add filters so certain trains will only stop at certain blocks in random mode.
- if a train comes in very fast and wants to stop at the mini block, the block is so small that it likely will overshoot the block quite a bit. You could fix that by setting the speeds on the markers from the larger block to "slow", so a train that will not stop at the large block will still slow down early enough.

I think a solution that would fit your issue better would be to allow for multiple markers within the same block, for different "sensor positions". This however would introduce a lot of complexity so I am unlikely to implement something like that in the foreseeable future.
